extensions/local_rate_limit/plugin.cc

```diff
@@ -0,0 +1,139 @@
+#include "extensions/local_rate_limit/plugin.h"
+
+#include "absl/strings/str_cat.h"
+#include "extensions/common/wasm/json_util.h"
+#include "extensions/local_rate_limit/bucket.h"
+
+using ::nlohmann::json;
+using ::Wasm::Common::JsonArrayIterate;
+using ::Wasm::Common::JsonGetField;
+using ::Wasm::Common::JsonObjectIterate;
+using ::Wasm::Common::JsonValueAs;
+
+namespace {
+
+// tooManyRequest returns a 429 response code.
+void tooManyRequest() { sendLocalResponse(429, "Too many requests", "", {}); }
```

Comment:
Since this is also instructional. add info about what was the recorded rps and what was the limit .

Reply:
But this will be returned to the client. I don't think we want to expose much detail to the client?

Reply:
https://tools.ietf.org/html/rfc6585#section-4
The rfc says that the response “SHOULD” include details of the condition.

Reply:
Updated to add a very response body. The envoy local rate limiter does similar: https://github.com/envoyproxy/envoy/blob/23a43b2eaf37f0a44a8953f076a0d055575b2a62/source/extensions/filters/http/local_ratelimit/local_ratelimit.cc#L99